airflow/operators/subdag_operator.py

```diff
@@ -157,4 +171,44 @@ def post_execute(self, context, result=None):
 
         if dag_run.state != State.SUCCESS:
             raise AirflowException(
-                "Expected state: SUCCESS. Actual state: {}".format(dag_run.state))
+                "Expected state: SUCCESS. Actual state: {}".format(dag_run.state)
+            )
+
+        if self.propagate_skipped_state and self._check_skipped_states(context):
+            self._skip_downstream_tasks(context)
+
+    def _check_skipped_states(self, context):
+        leaves_tis = self._get_leaves_tis(context['execution_date'])
+
+        if self.propagate_skipped_state == SkippedStatePropagationOptions.ANY_LEAF:
+            return any(ti.state == State.SKIPPED for ti in leaves_tis)
+        if self.propagate_skipped_state == SkippedStatePropagationOptions.ALL_LEAVES:
+            return all(ti.state == State.SKIPPED for ti in leaves_tis)
+        raise AirflowException(
```

Comment:
Should we validate this earlier in the `__init__`?

Reply:
Actually I only did the "validation" because pylint was complaining that an explicit return value is required and raising an exception made the most sense to me.